.github/workflows/on_pull_request.yaml

```diff
@@ -3,19 +3,24 @@ name: On Pull Request
 # On pull_request, we:
 # * always publish to charmhub at latest/edge/branchname
 # * always run tests
-
+# * always run builds with cache
 on:
   pull_request:
 
 jobs:
+  get-paths-and-build:
+    name: Get charm paths and build with cache
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/get_charms_build_with_cache.yaml@KF-6690-build-with-cache
```

Comment:
PR https://github.com/canonical/charmed-kubeflow-workflows/pull/88 has been approved so now we can remove the branch pin